### PR TITLE
Allow optional commands

### PR DESCRIPTION
**Pull request by otterdahl**
Issue: https://github.com/rsmp-nordic/rsmp_core/issues/148

### CODE REVIEW

**Comment by emiltin** [comment]
This requires support in the schema, simulator and validator.

File: source/conf.py [hunk 1]
```diff
@@ -57,9 +57,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '3.2.2'
+version = '3.3'
```
Comment (emiltin) [inline source/conf.py]: We should use 3.3.0

Comment (otterdahl) [reply source/conf.py]: Already updated in the RSMP 3.3 branch.

**Comment by marcgarba** [comment]
In "If a required argument is missing in CommandRequest it it be considered as a serious error resulting in MessageNotAck."
-> "If a required argument is missing in CommandRequest **it is** considered a serious error resulting in MessageNotAck."

**Comment by otterdahl** [comment]
> In "If a required argument is missing in CommandRequest it it be considered as a serious error resulting in MessageNotAck." -> "If a required argument is missing in CommandRequest **it is** considered a serious error resulting in MessageNotAck."

Thank you. Fixed

**Comment by marcgarba** [comment]
Thanks, but in others many paragraphs, I've seen nearly the same : "... this is considered a serious error resulting in MessageNotAck"
'it is **be** ...' or 'it is ...' or both possible? a detail...

**Comment by otterdahl** [comment]
> Thanks, but in others many paragraphs, I've seen nearly the same : "... this is considered a serious error resulting in MessageNotAck" 'it is **be** ...' or 'it is ...' or both possible? a detail...

Correct. `be` should be removed. Fixed

File: source/applicability/basic_structure.rst [hunk 2]
```diff
@@ -1191,8 +1191,10 @@ Command messages are used to give order using one or more commands, for the
 referenced object.
 The site responds with a command acknowledgement.
 
-All arguments needs to included in a command, otherwise it results a serious
-error resulting in MessageNotAck. See section about :ref:`incomplete-commands`.
+All arguments are considered required to be included in a CommandRequest unless
```
Comment (emiltin) [inline source/applicability/basic_structure.rst]: => All arguments in a CommandRequest are considered required unless

File: source/applicability/basic_structure.rst [hunk 3]
```diff
@@ -1191,8 +1191,10 @@ Command messages are used to give order using one or more commands, for the
...
+they are specifically marked as optional in the SXL. If a required argument is
+missing in CommandRequest it is considered as a serious error resulting in
```
Comment (emiltin) [inline source/applicability/basic_structure.rst]: => missing in a CommandRequest

File: source/applicability/basic_structure.rst (hunk 2 above)
Comment (otterdahl) [reply source/applicability/basic_structure.rst]: Fixed

File: source/applicability/basic_structure.rst (hunk 3 above)
Comment (otterdahl) [reply source/applicability/basic_structure.rst]: Fixed

**Pull request merged**